### PR TITLE
tests: consolidate boost xunit result files

### PR DESCRIPTION
**Pull request by avikivity**
The recent parallelization of boost unit tests caused an increase
in xml result files. This is challenging to Jenkins, since it
appears to use rpc-over-ssh to read the result files, and as a result
it takes more than an hour to read all result files when the Jenkins
main node is not on the same continent as the agent.

To fix this, merge the result files in test.py and leave one result
file per mode. Later we can leave one result file overall (integrating
the mode into the testsuite name), but that can wait.

Tested on a local Jenkins instance (just reading the result files,
not the entire build).

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -291,6 +291,8 @@ async def run(self, options):
     def print_summary(self):
         pass
 
+    def get_junit_etree(self):
+        return None
```
Comment (nyh) [inline test.py:295]: If I understand this code correctly (and I probably don't...) your code only works on the Boost tests. What about all the other tests which generate xunit result files, e.g., Alternator? Are those files just left alone and not "consolidated" with anything, and at the end their content is visible too?

Comment (avikivity) [reply test.py:295]: Right.

**Review by nyh — APPROVED** [review]

**Comment by avikivity** [comment]
I dequeued it: just noticed `boost.xml` won't be accepted by our files filter:

```
18:35:20  Going to publish xunit files: testlog/x86_64_release/xml/*.xunit.xml
```

**Comment by avikivity** [comment]
v2: changed output file name to ``boost.xunit.xml` to suit the filters used by our Jenkins scripts.

**Comment by avikivity** [comment]
Bump

**Comment by nyh** [comment]
Merged, again.
As in the first iteration I didn't know how to test this, so let's just pray that it will work.
Note that next hasn't been successfully promoted in 3 days, so even if this commit is broken, we won't know because something else is already broken :-( In any case, since you tested this yourself (and I guess the second time, you tested it even better), I decided to take this chance.

**Comment by avikivity** [comment]
Turns out it worked!

**Pull request closed without merging**